src/configuration.ts

```diff
@@ -285,17 +291,26 @@ export function getRelativePath(directory?: string | null, relativeTo?: string)
  * @return An array of absolute paths to directories potentially containing rules
  */
 export function getRulesDirectories(directories?: string | string[], relativeTo?: string): string[] {
-    const rulesDirectories = arrayify(directories)
-        .map((dir) => getRelativePath(dir, relativeTo))
+    return arrayify(directories)
+        .map((dir) => {
+            let absolutePath;
+            if (useAsPath(dir)) {
+                absolutePath = getRelativePath(dir, relativeTo);
+                if (absolutePath != null) {
+                    if (!fs.existsSync(absolutePath)) {
+                        throw new Error(`Could not find custom rule directory: ${dir}`);
+                    }
+                }
+            } else {
+                try {
+                    absolutePath = resolve.sync(dir, { basedir: relativeTo }).replace("index.js", "");
```

Comment:
just to be safe, because the path *may* include index.js earlier: `.replace(/index.js$/, "")`